compose/service/values/validator.go

```diff
@@ -26,38 +27,72 @@ type (
 	UniqueChecker    func(context.Context, store.Storer, *types.RecordValue, *types.ModuleField, *types.Module) (uint64, error)
 	ReferenceChecker func(context.Context, store.Storer, *types.RecordValue, *types.ModuleField, *types.Module) (bool, error)
 
+	localeService interface {
+		T(ctx context.Context, ns, key string, rr ...string) string
+	}
+
 	validator struct {
 		uniqueCheckerFn    UniqueChecker
 		recordRefCheckerFn ReferenceChecker
 		userRefCheckerFn   ReferenceChecker
 		fileRefCheckerFn   ReferenceChecker
 
+		localeSvc localeService
+
 		now func() time.Time
 	}
 )
 
+const (
+	ComposeResourceTranslationType = "compose"
+)
+
 func makeInternalErr(field *types.ModuleField, err error) types.RecordValueError {
-	return types.RecordValueError{Kind: "internal", Message: err.Error(), Meta: map[string]interface{}{"field": field.Name}}
+	return types.RecordValueError{
+		Kind:    "internal",
+		Message: err.Error(),
+		Meta:    map[string]interface{}{"field": field.Name},
+	}
 }
 
-func makeEmptyErr(field *types.ModuleField) types.RecordValueError {
-	return types.RecordValueError{Kind: "empty", Meta: map[string]interface{}{"field": field.Name}}
+func makeEmptyErr(ctx context.Context, field *types.ModuleField, ls localeService) types.RecordValueError {
+	return types.RecordValueError{
+		Kind:    "empty",
+		Meta:    map[string]interface{}{"field": field.Name},
+		Message: ls.T(ctx, ComposeResourceTranslationType, "record-field.errors.empty"),
```

Comment:
Value you used for translation namespace (`ComposeResourceTranslationType`) does not look right. Isn't this used for resource translations only? And you are not translating resources here (field labels on module for example) but static texts. 

See `RecordErrValueInput` and how translations are constructed there. Namespace is `compose`. See `src/en/corteza-server/compose/record.yaml` in `corteza-locale`.

Reply:
MB, I will fix the variable name, I introduced new var `ComposeResourceTranslationType = "compose"`.